### PR TITLE
[BIOMAGE-1159] Add sourcemaps to production build for easier debugging

### PR DESCRIPTION
**Pull request by marcellp**
# Background
#### Link to issue
https://biomage.atlassian.net/browse/BIOMAGE-1159 

#### Link to staging deployment URL 
https://ui-marcellp-ui373.scp-staging.biomage.net/

#### Links to any Pull Requests related to this
N/A

#### Anything else the reviewers should know about the changes here
N/A

# Changes
### Code changes
Refactored Next.js settings files and the Webpack configuration so we can have reliable source maps as well as React and Redux profilers in staging and production.

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [x] Unit tests written
- [x] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [x] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [x] Relevant Github READMEs updated
- [x] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [x] Photo of a cute animal attached to this PR
image

### CODE REVIEW

File: config/webpack/configRules.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        configFile: path.resolve('../.eslintrc'),
+        eslint: {
+          configFile: path.resolve(__dirname, '../.eslintrc'),
+        },
+      },
+      loader: 'eslint-loader',
+    });
+  }
+
+  // to be refactored
```
Comment (ivababukova) [inline config/webpack/configRules.js]: you mentioned on the call that this doesn't need refactoring anymore -- if this is indeed true, bettter remove the comment

File: config/webpack/configPlugins.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+const MomentLocalesPlugin = require('moment-locales-webpack-plugin');
+
+const webpackConfigPlugins = (config, { dev }) => {
+  const plugins = [
+    new webpack.IgnorePlugin(
+      /pages.*\/test.*/,
+    ),
+  ];
+
+  if (process.env.NODE_ENV === 'analyse') {
```
Comment (ivababukova) [inline config/webpack/configPlugins.js:14]: do does that mean that in order to use the analyse mode, people will need to set the `NODE_ENV` variable to `analyse`? Is there any action required by developers?

File: next.config.js [hunk 3]
```diff
@@ -45,198 +40,59 @@ if (typeof require !== 'undefined') {
...
-    use: {
-      loader: 'url-loader',
-      options: {
-        name: 'fonts/[hash].[ext]',
-        limit: 5000,
-        mimetype: 'application/font-woff',
+      {
+        source: '/experiments',
+        destination: '/data-management',
+        permanent: false,
```
Comment (ivababukova) [inline next.config.js:55]: why do we need this one as well?

**Review by ivababukova — CHANGES_REQUESTED** [review]
Code-wise looks good, I just have small questions. Make sure you add a section in the readme explaining how to get source maps working in staging and in production. Let me know after you've added the section, I will take a look again.

File: next.config.js (hunk 3 above)
Comment (marcellp) [reply next.config.js:55]: I guess in order to make the URL scheme more consistent. Right now when you click on a link to an experiment, it goes `/experiments/[experimentId]/data-exploration` for example. A user may try to access the URL under `/experiments`, assuming this is where they'd find their experiments. A redirect to data management makes sense for this. This was here before as you will see in the diff.

File: config/webpack/configPlugins.js (hunk 2 above)
Comment (marcellp) [reply config/webpack/configPlugins.js:14]: Technically yes, but we have a shorthand for it, it is `npm run analyze`, it is in the README. Mind you, `npm run analyze` only outputs a HTML file, other than that it behaves exactly like building the platform normally, so it's not like an additional feature that is enabled.

**Comment by marcellp** [comment]
Fixed.

File: README.md [hunk 4]
```diff
@@ -29,26 +29,34 @@ To get it running end-to-end locally with mocked dataset, you will need to set u
 
 Just follow the README of each of them for instructions on how to get it to run.
 
-### How to run tests in debug mode
+### Remote debugging with VS Code
```
Comment (ivababukova) [inline README.md:32]: it is worth adding a sentence that debugging remotely works also for staging and production urls and that for the debugger to work, the developer has to have the same version for the code of the ui locally, as it is for the environment they are trying to debug

File: config/webpack/configPlugins.js (hunk 2 above)
Comment (ivababukova) [reply config/webpack/configPlugins.js:14]: I can't see it in the readme, maybe you didn't push?

File: next.config.js (hunk 3 above)
Comment (ivababukova) [reply next.config.js:55]: Yes, I remember it was here, I was just wondering whether it is still needed. Sounds like it still is, let's leave it.

File: README.md (hunk 4 above)
Comment (marcellp) [reply README.md:32]: Fixed.

**Review by ivababukova — APPROVED** [review]
code looking good. After merging, make sure you show it to other people how to use it.

**Pull request merged**